lib/io.h

```diff
@@ -205,6 +243,12 @@ metal_io_phys_to_offset(struct metal_io_region *io, metal_phys_addr_t phys)
 static inline void *
 metal_io_phys_to_virt(struct metal_io_region *io, metal_phys_addr_t phys)
 {
+	metal_phys_addr_t tmp_phys = metal_io_phys_drv_to_phys(io, phys);
```

Comment:
Here you could create some regressions. This function is used to return a virtual address based on the physical address, not the remote physical address. This function is used for the buffers, but also the vrings (which use local physmap).

It seems that it would be better to provide a `metal_io_ops::phys_to_offset `that would point to a `metal_io_remote_phys_to_offset`or a platform specific  function, ensuring that the same IO region is not used for your vrings and the buffers in the vrings.